Riot/Modules/Home/HomeViewController.m

```diff
@@ -664,6 +671,108 @@ - (void)collectionView:(UICollectionView *)collectionView didSelectItemAtIndexPa
     [self.recentsSearchBar resignFirstResponder];
 }
 
+- (UIContextMenuConfiguration *)collectionView:(UICollectionView *)collectionView contextMenuConfigurationForItemAtIndexPath:(NSIndexPath *)indexPath point:(CGPoint)point API_AVAILABLE(ios(13.0))
+{
+    id<MXKRecentCellDataStoring> cellData = [recentsDataSource cellDataAtIndexPath:[NSIndexPath indexPathForRow:indexPath.item inSection:collectionView.tag]];
+    MXRoom *room = [self.dataSource getRoomAtIndexPath:[NSIndexPath indexPathForRow:indexPath.row inSection:collectionView.tag]];
+    NSString *roomId = room.roomId;
+    
+    return [UIContextMenuConfiguration configurationWithIdentifier:roomId previewProvider:^UIViewController * _Nullable {
+        // Add a preview using the cell's data to prevent the avatar and displayname from changing with a room list update.
+        return [[RoomPreviewViewController alloc] initWithCellData:cellData];
```

Comment:
I think we can make this a bit more generic:
* have it receive a snapshot UIView
* get the snapshot from the cell like `UIView *snapshotView = [[collectionView cellForItemAtIndexPath:indexPath] snapshotViewAfterScreenUpdates:NO];`
* maybe rename it to "ContextMenuItemPreviewViewController"
* simply add the snapshotView as a subview and drop the dependency on `RoomCollectionViewCell` and `MXKCellData`

Looks like this:
![Screenshot 2022-02-01 at 10 14 00](https://user-images.githubusercontent.com/637564/151933678-9513f929-0957-42a1-993a-1aa352ce23f4.png)

Reply:
Perhaps an even nicer option would be to pass the view to be snapshot directly to the view controller and let it do it. Maybe name it something like `ContextMenuSnapshotPreviewViewController` 🤔

Reply:
Niiiice! TIL about view snapshotting 🙌